tests/test_all_fields.py

```diff
@@ -0,0 +1,85 @@
+"""
+Test that with no fields selected for a stream automatic fields are still replicated
+"""
+
+from tap_tester import runner, menagerie
+from base import BaseTapTest
+
+class AllFieldsTest(BaseTapTest):
+    """Test that with no fields selected for a stream automatic fields are still replicated"""
+
+    # fields for which data is not generated
+    fields_to_remove = {
+        "worklogs": ["properties", "visibility"],
+        "project_types": ["icon"],
+        "users": ["name", "key", "expand"],
+        "versions": ["expand", "moveUnfixedIssuesTo", "project", "remotelinks", "operations"],
+        "issues": ["renderedFields", "schema", "editmeta", "fieldsToInclude", "versionedRepresentations", "names", "properties"],
+        "issue_comments": ["properties", "visibility", "renderedBody"],
+        "projects": ["roles", "url", "issueTypes", "email", "assigneeType", "components", "projectCategory"],
+        "issue_transitions": ["fields", "expand"],
+        "resolutions": ["iconUrl"],
+        "changelogs": ["historyMetadata"]
+    }
```

Comment:
@kspeer825 @KrisPersonal: IMO, we should CRUD records wherever possible, instead of skipping the field validation if records are empty for particular field. Ideally creating single record with all required field should resolve this issue.

Reply:
Rushi is correct, these fields should be covered where possible. If we cannot generate data for any fields the reasons need to be documented in the test on a per-field basis.

Reply:
@kspeer825 @RushT007 As per the discussion, we have made a backlog ticket for updating API calls for getting more fields and removed some fields from the fields_to_remove map in all_field_test for which data is generated.